src/app/film-list/film-list.component.html

```diff
@@ -15,4 +15,15 @@
   <film-card *ngFor="let film of films" [film]="film" [view]="selectedViewValue"></film-card>
 </div>
 
-<button (click)="addNextFilmsPage()" md-raised-button>Add more</button>
\ No newline at end of file
+<div [hidden]="isAddingNewFilms">
+  <button (click)="addNextFilmsPage()" 
+    md-raised-button [disabled]="isAddFilmsButtonDisabled()">Add more</button>
+</div>
+
+<div [hidden]="!isShownAddMoreProgressSpinner()">
+  <md-spinner></md-spinner>
+</div>
+
+<div class="go-up-btn">
+  <a md-mini-fab href="#top"><md-icon>arrow_upward</md-icon></a>
```

Comment:
it is made a directive to show or hide a DOM element when is scrolling.
the directive was done on base of your example. thanks for that!